### PR TITLE
 Metadata Collection

### PR DESCRIPTION
**Pull request by baerdal**
Adding metadata collection and comparison to all touchstone benchmarks

### CODE REVIEW

**Comment by baerdal** [comment]
Note: Requires new cpuinfo indexing scheme at https://github.com/cloud-bulldozer/scribe/pull/29

**Comment by baerdal** [comment]
Some sample runs and output:   @rsevilla87 @jtaleric 
vegeta
uperf

**Comment by baerdal** [comment]
With the tabulate module, we are able to get cleaner output that is easier to work with:
uperf

File: src/touchstone/utils/lib.py [hunk 1]
```diff
@@ -4,6 +4,14 @@
 _logger = logging.getLogger("touchstone")
 
 
+def print_metadata_dict(uuid, d):
+    for k, v in d.items():
+        if isinstance(v, dict):
+            print_metadata_dict(uuid, v)
+        else:
+            print("{0}, {1}, {2}, ".format(uuid, k, v))
```
Comment (rsevilla87) [inline src/touchstone/utils/lib.py:12]: Remove trailing comma

File: src/touchstone/utils/lib.py [hunk 2]
```diff
@@ -4,6 +4,14 @@
 _logger = logging.getLogger("touchstone")
 
 
+def print_metadata_dict(uuid, d):
+    for k, v in d.items():
```
Comment (rsevilla87) [inline src/touchstone/utils/lib.py:8]: We need to print a CSV header.
In addition, we also need to print the "where" field to identify the server/pod where the metadata was taken from. Resulting in something like below:

```
uuid, where, field, value 
b34fe2f0-9d40-553f-becc-3132d495f51e, uperf-server-0-80b78c16, value.Model name, Intel(R) Xeon(R) Platinum 8175M CPU @ 2.50GHz, 
b34fe2f0-9d40-553f-becc-3132d495f51e, uperf-server-0-80b78c16, value.Architecture, x86_64,                                                      
b34fe2f0-9d40-553f-becc-3132d495f51e, uperf-server-0-80b78c16, value.CPU(s), 4,   
etc.
```

**Pull request merged**